sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProjectTypeCollectorTest.java

```diff
@@ -0,0 +1,206 @@
+
+    assertThat(underTest.getSummary()).hasValue("Found 1 MSBuild project. 1 with no MAIN nor TEST files.");
+  }
+
+
+  @Test
+  public void withOnlyMainFiles() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+
+    // this will be skipped, considered as top-level module
+    addProjectWithNoFiles(underTest);
+
+    addMainProject(underTest);
+
+    assertThat(underTest.getSummary()).hasValue("Found 1 MSBuild project. 1 MAIN project.");
+  }
+
+  @Test
+  public void withOnlyTestFile() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+
+    // this will be skipped, considered as top-level module
+    addProjectWithNoFiles(underTest);
+
+    addTestProject(underTest);
+    addTestProject(underTest);
+    addTestProject(underTest);
+
+    assertThat(underTest.getSummary()).hasValue("Found 3 MSBuild projects. 3 TEST projects.");
+  }
+
+  @Test
+  public void withBothTypes_calledOnce_returnsEmpty() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+
+    // this will be skipped, considered as top-level module
+    addProjectWithNoFiles(underTest);
+
+    addMixedProject(underTest);
+
+    assertThat(underTest.getSummary()).hasValue("Found 1 MSBuild project. 1 with both MAIN and TEST files.");
+  }
+
+  @Test
+  public void mixedProjects_test_and_main() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+
+    // this will be skipped, considered as top-level module
+    addProjectWithNoFiles(underTest);
+
+    addTestProject(underTest);
+    addMainProject(underTest);
+    addTestProject(underTest);
+
+    assertThat(underTest.getSummary()).hasValue("Found 3 MSBuild projects. 1 MAIN project. 2 TEST projects.");
+  }
+
+  @Test
+  public void mixedProjects_test_and_both() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+
+    // this will be skipped, considered as top-level module
+    addProjectWithNoFiles(underTest);
+
+    addTestProject(underTest);
+    addMixedProject(underTest);
+    addTestProject(underTest);
+
+    assertThat(underTest.getSummary()).hasValue("Found 3 MSBuild projects. 2 TEST projects. 1 with both MAIN and TEST files.");
+  }
+
+  @Test
+  public void mixedProjects_main_and_both() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+
+    // this will be skipped, considered as top-level module
+    addProjectWithNoFiles(underTest);
+
+    addMainProject(underTest);
+    addMixedProject(underTest);
+    addMainProject(underTest);
+
+    assertThat(underTest.getSummary()).hasValue("Found 3 MSBuild projects. 2 MAIN projects. 1 with both MAIN and TEST files.");
+  }
+
+  @Test
+  public void mixedProjects_test_none() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+
+    // this will be skipped - considered as top-level module
+    addProjectWithNoFiles(underTest);
+
+    addTestProject(underTest);
+    addTestProject(underTest);
+
+    addProjectWithNoFiles(underTest);
```

Comment:
Other tests don't have empty line between adding. 
```suggestion
    addTestProject(underTest);
    addTestProject(underTest);
    addProjectWithNoFiles(underTest);
```